packages/query-core/src/hydration.ts

```diff
@@ -65,6 +66,7 @@ function dehydrateQuery(query: Query): DehydratedQuery {
     state: query.state,
     queryKey: query.queryKey,
     queryHash: query.queryHash,
+    ...(query.state.status === 'pending' && { promise: query.promise }),
```

Comment:
I'd recommend to "hide" the error here to prevent leaking secrets.
```suggestion
    ...(query.state.status === 'pending' && { promise: query.promise.catch(() => { throw new Error("redacted") }) }),
``` 

You might also want to consider different behaviour on rehydration of errors, depending of environment. I'd probably throw an error in SSR so the suspense boundary would just retry in the browser - and in the browser, I wouldn't throw at all, but immediately retry the query if it failed in RSC.

Reply:
thanks. How does this look?

03676ddd

Reply:
Looks good to me :)

Reply:
I think in Next.js (or maybe React does that), the error message and stack are already hidden when crossing the RSC boundary. This is also the case for Server Actions.

However, there will be a special `digest` field attached to the error which is a random ID. So developers can log the actual error on the server and use that ID from client logs to match it, without leaking anything.

https://nextjs.org/docs/app/building-your-application/routing/error-handling#securing-sensitive-error-information

Reply:
if react does this, it means we could remove this error handling. if it's next specific, I don't think we can rely on it because other frameworks like waku might implement this differently.

So can we figure out if this is next or react who does this ?

Reply:
I thought that was only for errors thrown during render, not for transported values, but I might be wrong - double-checking might definitely be a good idea here.

Reply:
Either way, this part is in the query core, so might make sense to redact it by default if entirely other frameworks handle it differently? Would be nice to support the digest too though. 🤔 

Easier to go from redacted -> more details in the future than the other way around so I think current version is fine to ship.

Reply:
true, I haven't even thought about other frameworks 👍

Reply:
@TkDodo @Ephem I confirmed that it's a React behavior (which makes sense, as it's done on the RSC encoding/decoding layer that frameworks can't inject into): https://github.com/facebook/react/blob/ea6e05912aa43a0bbfbee381752caa1817a41a86/packages/react-client/src/ReactFlightClient.js#L1477-L1484

If we redact it here, the client code (e.g. error boundaries, alert boxes, trackers) can't receive that digest value and get it displayed/logged.